Allow compiling without go-nvml support

DanielGibson opened the pull request:
@leahneukirchen reported that go-nvml can't be built on musl-based distros, see https://github.com/markusressel/fan2go/pull/373#issuecomment-2954120240

With these changes fan2go can be built without nvml-dependency (`make build-no-nvml`)

markusressel commented on internal/sensors/cmd.go:
```diff
@@ -22,6 +22,10 @@ func (sensor *CmdSensor) GetId() string {
 }
 
+func (sensor *CmdSensor) GetLabel() string {
+	return "CMD Sensor " + sensor.Config.ID
```
Why the prefixes?
The ID must be unique, which should be validated as well.
Main gripe is that now its inconsistent between fan types :thinking:

DanielGibson replied:
Well the usual fans have a label from the system like "hwmon3/fan1" so I thought in this case it makes sense to combine the type with the ID, so it doesn't look *exactly* the same as the ID.
However at least currently GetLabel() is not used for CMD or File fans or sensors (only used by `fan2go detect`), I just had to put *something* there so I could use the generic interface instead of `NvidiaFan`

markusressel commented:
Hi @DanielGibson , thx for the PR :heart: 
Looks good to me.

Is there a way for us to test/validate that the version compiler without nvml support actually works?

DanielGibson commented:
For testing, you can set the `LD_BIND_NOW` environment variable (to "1" or whatever, I think everything except for an empty string should work?). Then, without nvidia drivers, you get the same error as on NixOS.

However this was primarily meant to fix musl-libc based distros (like Alpine), not sure how to reproduce that particular problem on a "normal" (glibc-based) distro.

BTW, this most probably only fixes the fan2go executable itself, maybe the tests need the same treatment? Not sure how to tell `go test` to use that `disable_nvml` tag, though.

DanielGibson commented:
Fixed that, and also added a commit to stop fan2go (in "regular" mode when it's just controlling the fans) to spam the console with
```
 WARNING  Fan gpufan1 does not support PWM sensor reading, cannot check for third party changes to the PWM value
 WARNING  Fan gpufan2 does not support PWM sensor reading, cannot check for third party changes to the PWM value
```
I turned it from ui.Warning() into ui.Debug(), but maybe it should be removed entirely?
Fans not supporting reading PWM exist, and fan2go *could* warn about that once at startup or something, but I see no point in checking this (and complaining about it) every cycle

markusressel commented:
> Fans not supporting reading PWM exist, and fan2go could warn about that once at startup or something, but I see no point in checking this (and complaining about it) every cycle

Yes, although fan2go should be able to detect this and disable the "third party" check automatically. Printing a warning in this case would still be good, if the check isn't already disabled, but as you mentioned only once at (fan controller?) startup. If fan2go doesn't detect that the PWM cannot be read for whatever reason, the warning should still be printed, so the user can check if something is wrong, or simply disable the check manually in the config.

markusressel commented:
> Because https://github.com/markusressel/fan2go/commit/187ad2b179044dcfbf31c27fd9f80562bc00085b breaks the build-no-nvml case..

Sorry about that.

> However this was primarily meant to fix musl-libc based distros (like Alpine)

Hmm I still wonder how we can check this (that a build with `disable_nvml` doesn't crash) in the CICD :thinking:  Can we maybe use an alpine based docker container and run a build inside it or something? Because without this changes like the commit mentioned above can easily slip through as I am not working with this myself on a regular basis :/

DanielGibson commented:
The Github workflow runs the generated executables (`./dist/fan2go-linux-amd64 version`, `./dist/fan2go-linux-amd64-no-nvml version`), if the build without nvml would crash that should be detected there.

Maybe additionally that workflow could run `objdump -T ./dist/fan2go-linux-amd64-no-nvml | grep -c nvml` and ensure that it prints `0` - that should make sure that the build *really* was without nvml.
Not sure what one must execute to make a workflow run fail, though

DanielGibson commented:
Looks like a shell command exiting with code `1` (instead of `0` which means success) does not make the CI build fail :-/

DanielGibson commented:
Ok, now the CI build contains a check that the no-nvml version has no "nvml" symbols and that the regular version *does* have "nvml" symbols.
